vars/template.go

```diff
@@ -130,50 +129,81 @@ func (i interpolator) extractVarNames(value string) []string {
 	var names []string
 
 	for _, match := range interpolationRegex.FindAllSubmatch([]byte(value), -1) {
-		names = append(names, strings.TrimPrefix(string(match[1]), "!"))
+		names = append(names, string(match[1]))
 	}
 
 	return names
 }
 
-type varsLookup struct {
-	varsTracker
-}
+func parseVarName(name string) VariableReference {
+	var pathPieces []string
 
-var ErrEmptyVar = errors.New("empty var")
+	varRef := VariableReference{Name: name}
```

Comment:
I'm beginning to think that the Name shouldn't include Source, that way it'll just be Path + Fields and won't break all of our existing cred managers

Reply:
The problem is if we do `path` + `fields` we will need to concatenate them back with `.` and wrap them properly with `"`. I think the parsing of `(())` makes us to use ref.Path for creds manager.

And the logic in template will take care of looking up fields.

Reply:
We can probably use `parts[1]`
https://github.com/concourse/concourse/blob/46efd51a6657287ba07b78b5627fb880f68fc32e/vars/template.go#L144-L145

Reply:
> And the logic in template will take care of looking up fields.

True, I guess another question is how much processing should the cred manager do vs Concourse. They technically have all the information (e.g. Fields) required to fully resolve the secret, but it might be less code duplication if Concourse handles that part